gtsfm/data_association/data_assoc.py

```diff
@@ -107,16 +114,37 @@ def run(
                 raise RuntimeError("Some cameras must have been dropped ")
             triangulated_data.add_camera(cam)
 
-        logger.debug(
-            "[Data association] output number of tracks: %s",
-            triangulated_data.number_tracks(),
-        )
-        logger.debug(
-            "[Data association] output avg. track length: %s",
-            SfmResult(triangulated_data, None).get_track_length_statistics()[0],
-        )
-
-        return triangulated_data
+        mean_3d_track_length, median_3d_track_length, track_lengths_3d = SfmResult(triangulated_data, None).get_track_length_statistics()
+
+        logger.debug("[Data association] output number of tracks: %s", num_accepted_tracks)
+        logger.debug("[Data association] output avg. track length: %s", mean_3d_track_length)
+
+        # dump the 3d point cloud before Bundle Adjustment for offline visualization
+        points_3d = [list(triangulated_data.track(j).point3()) for j in range(num_accepted_tracks)]
+        data_assoc_metrics = {
+            "mean_2d_track_length": np.round(mean_2d_track_length,3),
+            "accepted_tracks_ratio": np.round(accepted_tracks_ratio,3),
+            "track_cheirality_failure_ratio": np.round(track_cheirality_failure_ratio,3),
+            "num_accepted_tracks": num_accepted_tracks,
+            "median_3d_track_length": median_3d_track_length,
+            "mean_3d_track_length": mean_3d_track_length,
+            "min_3d_track_length": int(track_lengths_3d.min()),
+            "max_3d_track_length": int(track_lengths_3d.max()),
+            "num_len_2_tracks": int(np.sum(track_lengths_3d == 2)),
```

Comment:
`np.histogram` might make the code cleaner?

Reply:
thanks, updated